qiskit/providers/fake_provider/fake_generic.py

```diff
@@ -0,0 +1,535 @@
+            num_qubits = self._supported_gates[inst].num_qubits
+            qarg_set = self._coupling_map if num_qubits > 1 else list(range(self.num_qubits))
+            if inst == "measure":
+                cmd_def.append(
+                    Command(
+                        name=inst,
+                        qubits=qarg_set,
+                        sequence=(
+                            self._get_calibration_sequence(inst, num_qubits, qarg_set)
+                            if self._calibrate_instructions
+                            else []
+                        ),
+                    )
+                )
+            else:
+                for qarg in qarg_set:
+                    qubits = [qarg] if num_qubits == 1 else qarg
+                    cmd_def.append(
+                        Command(
+                            name=inst,
+                            qubits=qubits,
+                            sequence=(
+                                self._get_calibration_sequence(inst, num_qubits, qubits)
+                                if self._calibrate_instructions
+                                else []
+                            ),
+                        )
+                    )
+
+        qubit_freq_est = np.random.normal(4.8, scale=0.01, size=self.num_qubits).tolist()
+        meas_freq_est = np.linspace(6.4, 6.6, self.num_qubits).tolist()
+        return PulseDefaults(
+            qubit_freq_est=qubit_freq_est,
+            meas_freq_est=meas_freq_est,
+            buffer=0,
+            pulse_library=_PULSE_LIBRARY,
+            cmd_def=cmd_def,
+        )
+
+    def _add_calibrations_to_target(self, inst_map: InstructionScheduleMap) -> None:
+        """Add calibration entries from provided pulse defaults to target.
+
+        Args:
+            inst_map (InstructionScheduleMap): pulse defaults with instruction schedule map
+
+        Returns:
+            None
+        """
+
+        # The calibration entries are directly injected into the gate map to
+        # avoid then being labeled as "user_provided".
+        for inst in inst_map.instructions:
+            for qarg in inst_map.qubits_with_instruction(inst):
+                try:
+                    qargs = tuple(qarg)
+                except TypeError:
+                    qargs = (qarg,)
+                # Do NOT call .get method. This parses Qobj immediately.
+                # This operation is computationally expensive and should be bypassed.
+                calibration_entry = inst_map._get_calibration_entry(inst, qargs)
+                if inst in self._target._gate_map:
+                    if inst == "measure":
+                        for qubit in qargs:
+                            self._target._gate_map[inst][(qubit,)].calibration = calibration_entry
+                    elif qargs in self._target._gate_map[inst] and inst not in ["delay", "reset"]:
+                        self._target._gate_map[inst][qargs].calibration = calibration_entry
```

Comment:
Is there a reason to not just do this at `add_instruction()` time, like you can set the calibration on the `InstructionProperties` object and that would avoid needing to interact with the target internals. Also if you needed to do this, you can just do `target[inst][qargs].calibration = calibration_entry`

Reply:
This method used to be public, that is the main reason why it was there, so that users could send their own calibrations. Now that it's not, we can remove it (done in [54496db](https://github.com/Qiskit/qiskit/pull/10266/commits/54496db3127ac6a754004a5f77d55770375ad016)). Still, not all calibrations can be added cleanly with `add_instruction()`. For `measure`, the instruction is added to all of the qubits in the target at once, but the calibrations need to be individually added qubit by qubit, else, they will not be detected later on. The new implementation, at least, doesn't interact with target internals  😬.